cspreports/models.py

```diff
@@ -99,6 +108,16 @@ def nice_report(self):
     def __str__(self):
         return self.nice_report
 
+    def full_clean(self):
+        super(CSPReport, self).full_clean()
+        for field_name in REQUIRED_FIELDS:
+            django_field_name = field_name.replace("-", "_")
+            if getattr(self, django_field_name) is None:
+                raise ValidationError(
+                    _('%(value)s should be a required field'),
+                    params={'value': django_field_name},
+                )
```

Comment:
I find this construction troublesome. The `full_clean` method should validate data to determine whether they are suitable to be stored in a database. Using `full_clean` to validate the CSP report itself, may prevent saving of invalid CSP reports, despite they may provide useful information.

Reply:
Thanks, that's really good feedback. I see your point of being misleading in the sense `full_clean` should validate data to determine whether they can be saved or not, I was mainly relying on the fact that `full_clean` method needs to be explicitly called (and its behavior handled). In our case though, this is not preventing invalid CSP reports to be saved, but they will be saved with an invalid flag set to `True`.

@adamalton any thoughts on this?

Reply:
`full_clean` are called from model forms. That could backfire pretty quickly.

Reply:
fair point! Would you recon writing a custom validation function (maybe a private one) could help?

Reply:
I'm actually thinking about custom model form. That could handle a lot of work.